userspace/libsinsp/container.h

```diff
@@ -38,16 +38,16 @@ limitations under the License.
 class sinsp_container_manager
 {
 public:
-	using map_ptr_t = const std::unordered_map<std::string, sinsp_container_info>*;
-	using entry_ptr_t = sinsp_container_info*;
+	using map_ptr_t = const std::unordered_map<std::string, std::shared_ptr<const sinsp_container_info>>*;
 
 	sinsp_container_manager(sinsp* inspector);
 	virtual ~sinsp_container_manager();
 
 	map_ptr_t get_containers();
```

Comment:
I don't think it will, but if we make the map multithreaded, we'll have to go back to pointers (or pointer-like objects) so we'll have to do the `s/->/./g` dance in the other direction. Once async CRI lands and we're happy with it, we can go back to this but for now I'd avoid the churn.